torch/testing/_internal/common_pruning.py

```diff
@@ -7,6 +7,22 @@
 from torch import nn
 
 
+def rows_are_subset(subset_tensor, superset_tensor) -> bool:
+    """
+    Checks to see if all rows in subset tensor are present in the superset tensor
```

Comment:
Preserved ordering of the subset tensor is the second assumption I was referring to - that is the basis of the speed-up compared to the original implementation.